src/engine/strat_engine/blockdev.rs

```diff
@@ -55,10 +56,67 @@ pub fn resolve_devices(paths: &[&Path]) -> io::Result<BTreeSet<Device>> {
     Ok(devices)
 }
 
+#[derive(Debug)]
+pub struct BlockDevs {
+    pub devs: BTreeMap<DevUuid, BlockDev>,
+}
+
+impl BlockDevs {
+    pub fn new(devs: BTreeMap<DevUuid, BlockDev>) -> BlockDevs {
+        BlockDevs { devs: devs }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.devs.is_empty()
+    }
+
+    /// Read the latest json across all blockdevs
+    pub fn read_metadata(&self) -> Option<Vec<u8>> {
+
+        let mut bds: Vec<&BlockDev> = self.devs
+            .iter()
+            .map(|(_, bd)| bd)
+            .filter(|bd| bd.bda.last_updated().is_some())
+            .collect();
+
+        bds.sort_by_key(|k| k.bda.last_updated().unwrap());
+
+        for bd in bds.iter().rev() {
+            let mut f = match OpenOptions::new()
+                .read(true)
+                .open(&bd.devnode) {
+                Ok(f) => f,
+                Err(_) => continue,
+            };
+            match bd.bda.load_state(&mut f) {
+                Ok(Some(json)) => return Some(json),
+                _ => continue,
```

Comment:
It seems troubling that this continues in the face of so many errors without any indication of the error. Perhaps the return type should be enriched. Especially in the second match it seems like the distinction between ```Ok(None)``` and ```Err(...)``` is kind of significant.